external-plugins/rehearse/plugin/main.go

```diff
@@ -0,0 +1,197 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"kubevirt.io/project-infra/external-plugins/rehearse/handler"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/test-infra/pkg/flagutil"
+	v1 "k8s.io/test-infra/prow/client/clientset/versioned/typed/prowjobs/v1"
+	"k8s.io/test-infra/prow/config/secret"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/git/v2"
+	"k8s.io/test-infra/prow/interrupts"
+
+	"kubevirt.io/project-infra/external-plugins/rehearse/server"
+)
+
+type options struct {
+	dryRun         bool
+	hmacSecretFile string
+	endpoint       string
+	port           int
+	prowConfigPath string
+	jobsConfigBase string
+	kubeconfig     string
+	jobsNs         string
+	alwaysRun      bool
+	cacheDir       string
+	github         prowflagutil.GitHubOptions
+}
+
+func (o *options) validate() {
+	var errs []error
+	if o.prowConfigPath == "" {
+		errs = append(errs, fmt.Errorf("prow-config-path can't be empty"))
+	}
+	if o.jobsConfigBase == "" {
+		errs = append(errs, fmt.Errorf("jobs-config-path can't be empty"))
+	}
+	if o.jobsNs == "" {
+		errs = append(errs, fmt.Errorf("jobs-namespace can't be empty"))
+	}
+	if o.cacheDir == "" {
+		errs = append(errs, fmt.Errorf("cache-dir can't be empty"))
+	}
+	err := o.github.Validate(o.dryRun)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	if len(errs) > 0 {
+		for _, err := range errs {
+			logrus.WithError(err).Error("entry validation failure")
+		}
+		logrus.Fatalf("Arguments validation failed!")
+	}
+}
+
+func gatherOptions() *options {
+	o := &options{}
+	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	fs.IntVar(&o.port,
+		"port",
+		9900,
+		"Port to listen on.")
+	fs.StringVar(&o.endpoint,
+		"endpoint",
+		"/",
+		"Endpoint to listen on.")
+	fs.BoolVar(&o.dryRun,
+		"dry-run",
+		false,
+		"If set, dump the job config to stdout.")
+	fs.StringVar(&o.hmacSecretFile,
+		"hmac-secret-file",
+		"/etc/webhook/hmac",
+		"Path to the file containing the GitHub HMAC secret.")
+	fs.StringVar(&o.kubeconfig,
+		"kubeconfig",
+		"",
+		"Path to kubeconfig. If empty, will try to use K8s defaults.")
+	fs.StringVar(&o.prowConfigPath,
+		"prow-config-path",
+		"",
+		"Path to Prow configuration (required).")
```

Comment:
Is it common to make flags required? Isn't that usually what positional arguments are for?

Reply:
You are correct, and I absolutely agree with you. Unfortunately, that's what this library supports and I prefer to use it because the other Prow components use it as well and we expose some default Prow flags through it.